core/dbt/contracts/state.py

```diff
@@ -41,7 +43,7 @@ def __init__(self, path: Path, current_path: Path):
                 exc.add_filename(str(sources_path))
                 raise
 
-        sources_current_path = self.current_path / "sources.json"
+        sources_current_path = self.target_path / "sources.json"
```

Comment:
for my own understanding - why isn't self.project_root considered for sources_current_path?

Reply:
Good catch. It was just an oversight on my part.